libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -103,28 +108,60 @@ internal class MapboxTripSession(
                         roadObjects = getRouteInitInfo(it)?.roadObjects ?: emptyList()
                     }
                     this@MapboxTripSession.primaryRoute = newPrimaryRoute
+                    logD("primary route update - finished", LOG_CATEGORY)
                 }
-                threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
-                    navigator.setAlternativeRoutes(routes.drop(1))
-                }
+                val updateAlternativesJob =
+                    threadController.getMainScopeAndRootJob().scope.launch(
```

Comment:
as `setRoutes` has become a suspend function I guess we can stop using `treadController` and start a new coroutine from the current scope:
 ```suggestion
                val updateAlternativesJob = launch(
```

Reply:
`suspend` functions do not have a coroutine scope so we cannot start additional coroutines directly from the `suspend` function. I guess we could try and avoid starting new coroutines and instead just change a dispatcher when necessary if the work is done sequentially anyway, but I'd leave that to a separate refactor.